code-studio/src/js/initApp/initApp.js

```diff
@@ -93,6 +95,23 @@ window.apps = {
         var lastServerResponse = reporting.getLastServerResponse();
         if (lastServerResponse.videoInfo) {
           showVideoDialog(lastServerResponse.videoInfo);
+        } else if (lastServerResponse.endOfStageExperience) {
+          let body = document.createElement('div');
+          let stageInfo = lastServerResponse.previousStageInfo;
+          let stageName = `${window.dashboard.i18n.t('stage')} ${stageInfo.position}: ${stageInfo.name}`;
```

Comment:
oh, indeed! I had assume that `const` did not have the advantage that `let` does of being block-scoped, so I was avoiding it for intentionally-scoped helper variables like these, but upon further research it appears that was entirely wrong. Gonna start using for everything now :D